### PR TITLE
Optimise some AnimationValue methods for size 🐉🐲

### PR DESCRIPTION
**Pull request by nox**
This shaves 40KB off of libxul.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20014)

### CODE REVIEW

**Comment by nox** [comment]
r? @emilio 

https://treeherder.mozilla.org/#/jobs?repo=try&revision=c17d74b3b23f579fc34a59f4a83be35efc54c1da

File: components/style/properties/helpers/animated_properties.mako.rs [hunk 1]
```diff
@@ -343,16 +344,20 @@ unsafe impl HasSimpleFFI for AnimationValueMap {}
...
     % for prop in data.longhands:
-        % if prop.animatable:
-            /// ${prop.name}
-            % if prop.is_animatable_with_computed_value:
-                ${prop.camel_case}(longhands::${prop.ident}::computed_value::T),
-            % else:
-                ${prop.camel_case}(<longhands::${prop.ident}::computed_value::T as ToAnimatedValue>::AnimatedValue),
-            % endif
-        % endif
+    % if prop.animatable:
```
Comment (emilio) [inline components/style/properties/helpers/animated_properties.mako.rs]: This condition can be simplified:

```
if prop.is_animatable_with_computed_value:
elif prop.animatable:
else:
endif
````

Not sure if it's worth, your call.

File: components/style/properties/data.py [hunk 2]
```diff
@@ -229,6 +229,11 @@ def explicitly_enabled_in_chrome(self):
     def enabled_in_content(self):
         return self.enabled_in == "content"
 
+    def base_type(self):
```
Comment (emilio) [inline components/style/properties/data.py]: s/base_type/specified_type?

File: components/style/properties/helpers/animated_properties.mako.rs (hunk 1 above)
Comment (emilio) [reply components/style/properties/helpers/animated_properties.mako.rs]: Ok it goes away later, oh well :)

File: components/style/properties/helpers/animated_properties.mako.rs [hunk 3]
```diff
@@ -566,35 +566,49 @@ fn animate_discrete<T: Clone>(this: &T, other: &T, procedure: Procedure) -> Resu
...
+        use self::AnimationValue::*;
+
+        Ok(unsafe {
+            #[repr(C)]
+            struct AnimationValueVariantRepr<T> {
+                tag: u16,
+                value: T
+            }
+
+            let this_tag = *(self as *const _ as *const u16);
```
Comment (emilio) [inline components/style/properties/helpers/animated_properties.mako.rs]: All this `tag` stuff could probably just call `id()`, wdyt?

**Review by emilio — APPROVED** [review]
;_;

r=me

File: components/style/properties/data.py (hunk 2 above)
Comment (nox) [reply components/style/properties/data.py]: `specified_type` is already a `Longhand` method that can return a boxed type.

File: components/style/properties/helpers/animated_properties.mako.rs (hunk 3 above)
Comment (nox) [reply components/style/properties/helpers/animated_properties.mako.rs]: Yeah but I just kept thing consistent with `PropertyDeclaration` and it makes the invariant needed for this code to be sound more explicit (that the discriminants must correspond in the representation).

**Comment by nox** [comment]
Oops, broken Tidy.

@bors-servo r=emilio

**Comment by CYBAI** [comment]
```
{"status": "CRASH", "group": "default", "message": null, "stack": null, "subtest": null, "test": "/html/webappapis/scripting/events/compile-event-handler-settings-objects.html", "line": 84353, "action": "test_result", "expected": "TIMEOUT"}
```

**Pull request merged**